### PR TITLE
Fix issue with creating HTTPS client connectors

### PR DESCRIPTION
**Pull request by pubudu91**
This PR fixes the above mentioned issue, brought up at [ballerina/issues/3258](https://github.com/ballerinalang/ballerina/issues/3258)

### CODE REVIEW

File: http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/common/Constants.java [hunk 1]
```diff
@@ -134,7 +134,9 @@
 
     public static final String PROTOCOL = "PROTOCOL";
 
-    public static final String PROTOCOL_NAME = "http";
+    public static final String PROTOCOL_HTTP = "http";
```
Comment (lankavitharana) [inline http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/common/Constants.java]: this is scheme right?

File: http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/message/HTTPMessageUtil.java [hunk 2]
```diff
@@ -131,14 +131,18 @@ private static CarbonMessage setWebSocketCommonProperties(CarbonMessage cMsg,
      * Extract sender configuration from transport configuration.
      *
      * @param transportsConfiguration {@link TransportsConfiguration} which sender configurations should be extracted.
+     * @param isHttps
      * @return extracted {@link SenderConfiguration}.
      */
-    public static SenderConfiguration getSenderConfiguration(TransportsConfiguration transportsConfiguration) {
+    public static SenderConfiguration getSenderConfiguration(TransportsConfiguration transportsConfiguration,
```
Comment (lankavitharana) [inline http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/message/HTTPMessageUtil.java]: shall we get the scheme here and check it here rather than checking it in ballerina side?

File: http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/message/HTTPMessageUtil.java [hunk 3]
```diff
@@ -131,14 +131,18 @@ private static CarbonMessage setWebSocketCommonProperties(CarbonMessage cMsg,
      * Extract sender configuration from transport configuration.
      *
      * @param transportsConfiguration {@link TransportsConfiguration} which sender configurations should be extracted.
+     * @param scheme
      * @return extracted {@link SenderConfiguration}.
      */
-    public static SenderConfiguration getSenderConfiguration(TransportsConfiguration transportsConfiguration) {
+    public static SenderConfiguration getSenderConfiguration(TransportsConfiguration transportsConfiguration,
```
Comment (chanakaudaya) [inline http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/message/HTTPMessageUtil.java:137]: This method does not fit to the intention of this class (HTTPMessageUtil). Shall we move these transport configuration related methods to a separate utility class?

File: http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/util/server/HttpsServer.java [hunk 4]
```diff
@@ -75,7 +75,7 @@ public void start() {
             KeyManagerFactory kmf = KeyManagerFactory.getInstance("SunX509");
             kmf.init(ks, ctPass);
 
-            sslContext = SSLContext.getInstance("TLSv1.2");
+            sslContext = SSLContext.getInstance("TLS");
```
Comment (chanakaudaya) [inline http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/util/server/HttpsServer.java:78]: Why is this changed to TLS from TLSv1.2?

Comment (pubudu91) [reply http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/util/server/HttpsServer.java:78]: To prevent this from being specific to a particular TLS version

File: http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/message/HTTPMessageUtil.java (hunk 3 above)
Comment (pubudu91) [reply http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/message/HTTPMessageUtil.java:137]: What if we change the class name to something broader? i.e: HTTPUtils?

File: http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/common/Constants.java [hunk 5]
```diff
@@ -134,7 +134,9 @@
 
     public static final String PROTOCOL = "PROTOCOL";
 
-    public static final String PROTOCOL_NAME = "http";
+    public static final String SCHEME_HTTP = "http";
+
+    public static final String SCHEME_HTTPS = "https";
```
Comment (shafreenAnfar) [inline http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/common/Constants.java]: On second thoughts, would it sound better if we say HTTP_SCHEME instead of SCHEME_HTTP ?

File: http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/https/HTTPSClientTestCase.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+/**
+ * Tests for HTTPS client connector
+ */
+public class HTTPSClientTestCase {
+
+    private final Logger log = LoggerFactory.getLogger(HTTPSClientTestCase.class);
+
+    private HttpsServer httpsServer;
+    private HttpClientConnector httpClientConnector;
+    private int port = 9092;
```
Comment (shafreenAnfar) [inline http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/https/HTTPSClientTestCase.java]: Should we add it to TestUtil as a constant and then refer to it ?

Comment (pubudu91) [reply http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/https/HTTPSClientTestCase.java]: Ack. Will use the HTTPS port defined in TestUtil

File: http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/message/HTTPConnectorUtil.java [hunk 7]
```diff
@@ -191,7 +195,7 @@ public static ListenerConfiguration buildListenerConfig(String id, Map<String, S
                 properties.get(Constants.HTTP_HOST) : Constants.HTTP_DEFAULT_HOST;
         int port = Integer.parseInt(properties.get(Constants.HTTP_PORT));
         ListenerConfiguration config = new ListenerConfiguration(id, host, port);
-        String schema = properties.get(Constants.HTTP_SCHEME);
+        String schema = properties.get(Constants.SCHEME);
```
Comment (shafreenAnfar) [inline http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/message/HTTPConnectorUtil.java:198]: Variable name is wrong.

File: http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/https/HTTPSClientTestCase.java [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+/**
+ * Tests for HTTPS client connector
+ */
+public class HTTPSClientTestCase {
+
+    private final Logger log = LoggerFactory.getLogger(HTTPSClientTestCase.class);
+
+    private HttpsServer httpsServer;
+    private HttpClientConnector httpClientConnector;
+//    private int port = 9092;
```
Comment (shafreenAnfar) [inline http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/https/HTTPSClientTestCase.java:55]: Shall we remove the commented code ?

**Review by shafreenAnfar — APPROVED** [review]

File: http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/https/HTTPSClientTestCase.java [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+            responseFuture.setHttpConnectorListener(listener);
+
+            latch.await(5, TimeUnit.SECONDS);
+
+            HTTPCarbonMessage response = listener.getHttpResponseMessage();
+            assertNotNull(response);
+            String result = new BufferedReader(new InputStreamReader(response.getInputStream()))
+                    .lines().collect(Collectors.joining("\n"));
+
+            assertEquals(testValue, result);
```
Comment (ruwanta) [inline http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/https/HTTPSClientTestCase.java:91]: Looks like the variables are switched.
public static void assertEquals(Object actual, Object expected) {
